dbt/include/spark/macros/adapters.sql

```diff
@@ -160,6 +163,53 @@
 {%- endmacro -%}
 
 
+{% macro persist_constraints(relation, model) %}
+  {{ return(adapter.dispatch('persist_constraints', 'dbt')(relation, model)) }}
+{% endmacro %}
+
+{% macro spark__persist_constraints(relation, model) %}
+  {% if config.get('constraints_enabled', False) and config.get('file_format', 'delta') == 'delta' %}
+    {% do alter_table_add_constraints(relation, model.columns) %}
+    {% do alter_column_set_constraints(relation, model.columns) %}
+  {% endif %}
+{% endmacro %}
+
+{% macro alter_table_add_constraints(relation, constraints) %}
+  {{ return(adapter.dispatch('alter_table_add_constraints', 'dbt')(relation, constraints)) }}
+{% endmacro %}
+
+{% macro spark__alter_table_add_constraints(relation, column_dict) %}
+
+  {% for column_name in column_dict %}
+    {% set constraints_check = column_dict[column_name]['constraints_check'] %}
+    {% for constraint_check in constraints_check %}
```

Comment:
In the current implementation, `constraints_check` is a string, not a list, so we shouldn't loop over it here. That's why we're only seeing the first character (`(`) show up in [the CI test](https://app.circleci.com/pipelines/github/dbt-labs/dbt-spark/1859/workflows/09685158-ba76-494c-9efc-3d1bb290731c/jobs/8365?invite=true#step-102-1216)!
```sql
alter table test16752556745428623819_test_constraints.my_model add constraint 50006a0485dbd5df8255e85e2c79411f check (id ();
```
We are planning to change this in the future by unifying these into a single `constraints` attribute: https://github.com/dbt-labs/dbt-core/issues/6750

Reply:
I was just looking at it, yes, and I will update my code.
In my mind it could (or should) be a list. I am not sure why we would only allow 1 value. (maybe worth considering for the future)

Reply:
Agreed! It should be a list. TK in 6750